src/goto-programs/write_goto_binary.cpp

```diff
@@ -148,22 +150,18 @@ bool write_goto_binary(
   irep_serializationt::ireps_containert irepc;
   irep_serializationt irepconverter(irepc);
 
-  switch(version)
-  {
-  case 1:
-  case 2:
-  case 3:
-    throw "version no longer supported";
-
-  case 4:
+  const int current_goto_version = 4;
+  if(version < current_goto_version)
+    throw invalid_user_input_exceptiont(
+      "version no longer supported",
+      "supported version = " + std::to_string(current_goto_version));
+  else if(version > current_goto_version)
```

Comment:
If you print the current version, also print the version of the goto binary.

Reply:
Done.